kubernetes/cray-nexus/templates/hooks.yaml

```diff
@@ -1,3 +1,65 @@
+#
+# MIT License
+#
+# (C) Copyright 2020-2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+{{- if .Values.security.keycloak.enabled }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "cray-nexus.fullname" . }}-secret-check
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "cray-nexus.labels" . | indent 4 }}
+  annotations:
+    helm.sh/hook: pre-install, pre-upgrade
+    helm.sh/hook-delete-policy: hook-succeeded
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: "{{ .Values.hooks.preInstall.serviceAccountName }}"
+      restartPolicy: Never
+      containers:
+        - name: secret-check
+          image: "{{ .Values.hooks.preInstall.image.repository }}:{{ .Values.hooks.preInstall.image.tag }}"
+          imagePullPolicy: "{{ .Values.hooks.preInstall.image.pullPolicy }}"
+          command:
+          - /bin/sh
+          - -c
+          - |-
+            set -x
+
+            # Wait for secret to exist in nexus namespace
+            while [[ ! $(kubectl get secrets -n nexus | awk '{print $1}' | grep {{ .Values.security.keycloak.client.secret.name }}) ]]; do
+                echo >&2 "Looking for secret again in 10 seconds"
```

Comment:
Specify the specific secret you are looking for. That way the user will understand why this is failing. Also, you could say that this secret is installed by the keycloak chart.

Reply:
Okay.  I had it that way, but then I changed it to use the same secret value as the `keycloak-realm.yaml` so this hook is waiting on the same secret as the code that needs it.